### DISCUSSION TITLE
The problem of "variance" in flow field inversion

### DISCUSSION THREAD

**Discussion by 15175829310** [mdolab/dafoam#683] [General]
What is the problem with displaying this error. thank。
![error](https://github.com/user-attachments/assets/f72fbfd8-4e65-404f-9c01-1fc67af22f9e)
[logOpt.txt](https://github.com/user-attachments/files/17008172/logOpt.txt)

**Comment by NeuroNexusAI** [comment]
To assist with the error you're experiencing, please provide more details about the error message and the context in which it appears. This will help in diagnosing and addressing the issue effectively.

**Reply by 15175829310** [reply]
Adding cells with centre within boxes 1((-5 -5 -5) (5 5 5))
Find 0 reference points for variance of U


--> FOAM FATAL ERROR: 
varData field exists but one can not find any valid data!

**Comment by friedenhe** [comment]
You saw this error because DAFoam can not find a cell that fully contains your prescribed probe point coordinates. I suggest you start with a single probe point and make sure the prescribed probe point coordinates are completely within a cell.

**Reply by 15175829310** [reply]
Thank you for your reply. I use the command 'postProcess func write CellCenters' to export coordinates and then select the points I want, but the problem shown above is strange. I look forward to your help.

**Reply by 15175829310** [reply]
I tried 1600 points this time, and in the 'logOpt. txt' file, it initially displayed 'Find 1600 reference points for variance of U', but later it displayed 'Find 1 reference points for variance of U.' May I ask which one should be used and why this situation occurs.
![1600](https://github.com/user-attachments/assets/c48021b3-e3c8-458c-99ce-3f01d9c0d665)
![1](https://github.com/user-attachments/assets/b6bf1c05-667a-4d16-b53a-e5bf9b5f9838)
[logOpt.txt](https://github.com/user-attachments/files/17082413/logOpt.txt)

**Comment by friedenhe** [comment]
This is very strange... We did not have this problem before.. Can you do a quick test to check if this behavior depends on the number of probe points? For example, if you reduce the number of probe points from 1600 to 100, will you still have this issue?

**Reply by 15175829310** [reply]
Yes, this problem still occurs. I tried to use 60 points, but during the calculation process, it showed only 46 points, which is strange

**Comment by friedenhe** [comment]
Have you solved the problem? Have you tried to optimize the case and see if the objective is indeed improved (e.g., velocity difference between the trained CFD simulations and ref data)?

**Reply by 15175829310** [reply]
1. I haven't solved this problem yet.
2. Although the number of points calculated is different from what I set, the optimization still has an effect

**Reply by 15175829310** [reply]
And I found that there were 100 reference points set in the Ramp example, but in the end, there were only 84

**Comment by friedenhe** [comment]
This could be a potential bug in the `DAUtility::myFindCell` function. We do not have time to debug this at this moment. A temporally solution is to use the `mesh.findNearestCell` function instead. You need to replace  `DAUtility::myFindCell`  with `mesh.findNearestCell` in DAObjFuncVariance.C and recompile
